Add SECURITY.md (#3643)
apache/parquet-java: SECURITY.md

<!--
  ~ Licensed to the Apache Software Foundation (ASF) under one
  ~ or more contributor license agreements.  See the NOTICE file
  ~ distributed with this work for additional information
  ~ regarding copyright ownership.  The ASF licenses this file
  ~ to you under the Apache License, Version 2.0 (the
  ~ "License"); you may not use this file except in compliance
  ~ with the License.  You may obtain a copy of the License at
  ~
  ~   http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing,
  ~ software distributed under the License is distributed on an
  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  ~ KIND, either express or implied.  See the License for the
  ~ specific language governing permissions and limitations
  ~ under the License.
  -->

# Security

## Reporting a Vulnerability

> Please report suspected vulnerabilities privately to the Parquet Project Management

Author: ```suggestion
Please report suspected vulnerabilities privately to the
```

> ASF security  at security@apache.org. Do **not** file a public GitHub issue or pull

Author: ```suggestion
ASF security team at security@apache.org (they will triage and coordinate the Parquet PMC). 
Do **not** file a public GitHub issue or pull
```

> Committee at private@parquet.apache.org. Do **not** file a public GitHub issue or pull

Author: ```suggestion
ASF security  at security@apache.org. Do **not** file a public GitHub issue or pull
```

> request for a suspected vulnerability, as that would disclose it before a fix is available.

Reviewer: The security contact should start with `security@`. There are two kinds of PMCs at the ASF:

- **PMCs with their own project-specific security mailing list** (e.g. Airflow, Tomcat, etc.), where a subset of committers handles vulnerabilities directly. These teams often have more bandwidth than the ASF Security Team, so we generally don't touch reports for them, unless a report is misdirected to `security@apache.org`, in which case we apply a "false positive filter" to avoid forwarding spam.
- **PMCs that rely on the main ASF Security Team.** For these, we expect all security reports to go through `security@apache.org`, and we only forward reports that are VALID according to the project's threat model, or that reveal a gap in it.

If you'd like your own list, you can request a [project-specific security mailing list](https://apache.org/security/committers.html#lists). Either way, I'd prefer `private@parquet.apache.org` not be used as the security contact.

Author: updated to point to security@apache.org thanks.

### Class loading

Several modules (e.g. parquet-avro, parquet-protobuf, parquet-thrift, etc.) provide functionality to translate Parquet data into Java objects. In the general case, this requires loading classes named in metadata stored in the Parquet files.

As a consequence:
 * When using these translation layers it is the user's responsibility to make sure their classpath is secure and does not contain any classes that when loaded might cause an adverse issue for their system.
 * If class loading driven by file contents is problematic, end users should use APIs to directly read the data from Parquet instead of relying on translation.
 * Arbitrary class loading by itself is not considered a security issue by the library. A security issue must demonstrate further ability to execute

>    arbitrary methods on the instantiated object.

Author: ```suggestion
   arbitrary methods on the instantiated object in such a way that it presents a security risk.
```
